File: src/MedicalCheckupF/Person.fs
namespace MedicalCheckupF

module PersonInformation =
    type Person =
        { Id: int
          Name: string
          Age: int
          Height: float
          Weight: float
          Ldl: int
          Hdl: int }
    
    let getBmi (person: Person) : float = person.Weight / person.Height ** 2
    let getTotalCholesterol (person: Person) : float = float (person.Hdl + person.Ldl)
    let setWeight (person: Person) (weight: double) : Person = { person with Weight = weight }
    let getRiskScore (person: Person) : float =
        getBmi person + getTotalCholesterol person

File: src/MedicalCheckup/Person.cs
namespace MedicalCheckup;

public class Person
{
    private int Id { init; get; }

    private string Name { init; get; }

    // In Meters.
    private double Height { get; }

    // In KG.
    private double Weight { get; set; }
    private int Ldl { init; get; }
    private int Hdl { init; get; }

    public Person(int id,
                  string name,
                  double height,
                  double weight,
                  int ldl,
                  int hdl)
    {
        Id = id;
        Name = name;
        Height = height;
        Weight = weight;
        Ldl = ldl;
        Hdl = hdl;
    }

    public double GetBmi() => Weight / Math.Pow(Height, 2);
    public double GetTotalCholesterol() => Hdl + Ldl;
    
    // Mutation is allowed, but it is encapsulated.
    public void SetWeight(double weight)
    {
        Weight = weight;
    }
}